client/src/test/java/com/sinch/sdk/e2e/domains/voice/CallsSteps.java

```diff
@@ -0,0 +1,134 @@
+                        .setLocale("en-US")
+                        .build()))
+            .setAction(ActionHangUp.builder().build())
+            .build();
+    service.update("1ce0ffee-ca11-ca11-ca11-abcdef000022", request);
+    updatePassed = true;
+  }
+
+  @When("^I send a request to update a call that doesn't exist$")
+  public void updateCallNotExits() {
+
+    SVAMLControl request =
+        SVAMLControl.builder()
+            .setInstructions(
+                Arrays.asList(
+                    InstructionSay.builder()
+                        .setText("Sorry, the conference has been cancelled. The call will end now.")
+                        .setLocale("en-US")
+                        .build()))
+            .setAction(ActionHangUp.builder().build())
+            .build();
+    try {
+      service.update("not-existing-callId", request);
+    } catch (ApiException ae) {
+      updateNotFoundException = ae;
+    }
+  }
+
+  @When("^I send a request to manage a call with callLeg$")
+  public void manageCallWithCallLeg() {
+
+    SVAMLControl request =
+        SVAMLControl.builder()
+            .setInstructions(
+                Arrays.asList(
+                    InstructionPlayFiles.builder()
+                        .setIds(
+                            Arrays.asList(
+                                "https://samples-files.com/samples/Audio/mp3/sample-file-4.mp3"))
+                        .build()))
+            .setAction(ActionContinue.builder().build())
+            .build();
+    service.manageWithCallLeg("1ce0ffee-ca11-ca11-ca11-abcdef000022", CallLegType.CALLEE, request);
+    manageWithCallLegPassed = true;
+  }
+
+  @Then("the response contains the information about the call")
+  public void getCallResult() {
+    CallInformation information =
+        CallInformation.builder()
+            .setTo(DestinationNumber.valueOf("+12017777777"))
+            .setDomain(DomainType.PSTN)
+            .setCallId("1ce0ffee-ca11-ca11-ca11-abcdef000003")
+            .setDuration(14)
+            .setStatus(CallStatusType.FINAL)
+            .setResult(CallResultType.ANSWERED)
+            .setReason(CallReasonType.MANAGER_HANGUP)
+            .setTimeStamp(Instant.parse("2024-06-06T17:36:00Z"))
+            .setCustom("Custom text")
+            .setUserRate(Price.builder().setCurrencyId("EUR").setAmount(0.1758F).build())
+            .setDebit(Price.builder().setCurrencyId("EUR").setAmount(0.1758F).build())
+            .build();
+    TestHelpers.recursiveEquals(getResponse, information);
+  }
+
+  @Then("the update call response contains no data")
+  public void updateCallResult() {
+    Assertions.assertEquals(true, updatePassed);
+  }
+
+  @Then("the update call response contains a \"not found\" error")
+  public void updateCallNotFoundResult() {
+    Assertions.assertTrue(updateNotFoundException.getMessage().contains("not found"));
```

Comment:
No. It will require dedicated Exception classes (we have an opened ADR for this: 0005-unified-exceptions)

Currently underlying error are used to create the ApiException with dedicated message adapted from error content (https://github.com/sinch/sinch-sdk-java/blob/main/core/src/main/com/sinch/sdk/core/exceptions/ApiExceptionBuilder.java)